helm-quarry/templates/web_deployment.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: web
+  labels:
+    app: web
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: web
+  template:
+    metadata:
+      labels:
+        app: web
+    spec:
+      containers:
+        - name: web
+          image: {{ .Values.web.repository }}:{{ .Values.web.tag }}
+{{ if .Values.localdev.enabled }}
+          imagePullPolicy: Never
+{{ else }}
+          imagePullPolicy: Always
+{{ end }}
+          command: ["python"]
+          args: ["quarry.wsgi"]
+          volumeMounts:
+            - mountPath: "/results"
+              name: results
+            - name: config
+              mountPath: /config/
+      volumes:
+        - name: results
+          persistentVolumeClaim:
+            claimName: results
+        - name: config
+          configMap:
+            name: config
```

Comment:
Yeah this is the place for it. Looks like the web system is using about 5% of it's two CPUs and, 1400M memory. So maybe request about 200m cpu, and 2G memory? I don't think I would add a limit until something became a problem. If it started running away with resources horizontal pod autoscaling may be a remedy in that case.

As it stands do you think that the web container could autoscale? Is there any state in the process itself? Or does it all reference what is happening in the DB? I think the workers could scale...though that would take a little work for the scale in, as the celery runner would need to be complete before finishing, though I believe they have a timeout so the grace period could probably just be set to that and all would work fine.

Reply:
Since the cluster is only running a single application, maybe be more liberal with requests? But I'm no SRE engineer, so let's go with whatever you think!

Reply:
Let's see if we can make quarry a nice micro service. I'm going to set it low, with a fair number of replicas. Low requests will help with autoscaling and the stability of having many.